uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/infra/LayoutEditorCssHelper.java

```diff
@@ -0,0 +1,235 @@
+        String propValue = propertyMap.get(property.getName());
+        return createPropertyField(property, propValue == null ? "" : propValue);
+    }
+
+    public PropertyEditorFieldInfo createPropertyField(CssProperty property, String value) {
+        String displayName = propertyNameMapI18n.get(property);
+        String helpText = propertyHelpMapI18n.get(property);
+        PropertyEditorType propType = resolveFieldType(property);
+        List<PropertyFieldValidator> validators = getValidators(property);
+        List<CssAllowedValue> allowedValues = property.getAllowedValues();
+        CssAllowedValue allowedValue = parseCssAllowedValue(property, value);
+        String currentValue = allowedValue != null ? formatCssAllowedValue(property, allowedValue) : value;
+
+        PropertyEditorFieldInfo field = new PropertyEditorFieldInfo(displayName, currentValue, propType)
+                .withKey(property.getName());
+
+        if (helpText != null) {
+            field.withHelpInfo("", helpText);
+        }
+        if (validators != null && !validators.isEmpty()) {
+            field.withValidators(validators);
+        }
+        if (allowedValues != null && !allowedValues.isEmpty()) {
+            field.withComboValues(allowedValues.stream()
+                    .map(v -> formatCssAllowedValue(property, v))
+                    .collect(Collectors.toList()));
+        }
+
+        return field;
+    }
+
+    public String formatCssAllowedValue(CssProperty property, CssAllowedValue value) {
+        String i18nKey = property.toString() + "__" + value.toString();
+        return CssAllowedValueConstants.INSTANCE.getString(i18nKey);
+    }
+
+    public CssAllowedValue parseCssAllowedValue(String property, String value) {
+        CssProperty cssProperty = CssProperty.get(property);
+        if (cssProperty != null) {
+            return parseCssAllowedValue(cssProperty, value);
+        }
+        return null;
+    }
+
+    public CssAllowedValue parseCssAllowedValue(CssProperty property, String value) {
+        for (CssAllowedValue cssAllowedValue : property.getAllowedValues()) {
+            String targetValue = formatCssAllowedValue(property, cssAllowedValue);
+            if (value.equals(targetValue)) {
+                return cssAllowedValue;
+            }
+        }
+        return null;
+    }
+
+    public List<CssValue> readCssValues(Map<String,String> propertyMap) {
+        if (propertyMap == null || propertyMap.isEmpty()) {
+            return new ArrayList<>();
+        }
+
+        return propertyMap.entrySet().stream()
+            .filter(entry -> CssProperty.get(entry.getKey()) != null)
+            .map(entry -> new CssValue(entry.getKey(), entry.getValue()))
+            .collect(Collectors.toList());
+    }
+
+    // Validator classes
+
+    public PropertyFieldValidator LENGTH_VALIDATOR = new PropertyFieldValidator() {
```

Comment:
extract to another class?

Reply:
Done => https://github.com/dgutierr/appformer/blob/926a6458c55e548441d99f2a311c3415a2234f43/uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/validator/LayoutEditorPropertyLengthValidator.java